browser/js/calculator.js

```diff
@@ -22,229 +28,247 @@
-      || keyValue === buttons.MULTIPLY
+  Calculator.prototype.isOperator = function ( keyValue ) {
+    return keyValue === Calculator.buttons.ADD
+      || keyValue === Calculator.buttons.SUBTRACT
+      || keyValue === Calculator.buttons.DIVIDE
+      || keyValue === Calculator.buttons.MULTIPLY
   }
-  function isEquals( keyValue ) {
-    return keyValue === buttons.ENTER || keyValue === buttons.EQUALS
+  Calculator.prototype.isEquals = function ( keyValue ) {
+    return keyValue === Calculator.buttons.ENTER || keyValue === Calculator.buttons.EQUALS
   }
-  function isClear( keyValue ) {
-    return keyValue === buttons.CLEAR
-      || keyValue === buttons.BACKSPACE
-      || keyValue === buttons.DELETE
+  Calculator.prototype.isClear = function ( keyValue ) {
+    return keyValue === Calculator.buttons.CLEAR
+      || keyValue === Calculator.buttons.BACKSPACE
+      || keyValue === Calculator.buttons.DELETE
   }
-  function isDot( keyValue ) {
-    return keyValue === buttons.DOT
+  Calculator.prototype.isDot = function ( keyValue ) {
+    return keyValue === Calculator.buttons.DOT
   }
-  function isNumber( keyValue ) {
+  Calculator.prototype.isNumber = function ( keyValue ) {
     return parseInt( keyValue ) || keyValue === '0'
   }
-  function isGreyButton( keyValue ) {
-    return keyValue === buttons.CLEAR
+  Calculator.prototype.isGreyButton = function ( keyValue ) {
+    return keyValue === Calculator.buttons.CLEAR
       || parseInt( keyValue )
       || keyValue === '0'
-      || keyValue === buttons.DOT
+      || keyValue === Calculator.buttons.DOT
   }
-  function fakeClickEffect( button, color ) {
-    getButton( button )
+  Calculator.prototype.fakeClickEffect = function ( button, color ) {
+    //FIXME: ghetto binding
+    var me = this
+    this.getButton( button )
       .classList.add( color )
     setTimeout( function() {
-      getButton( button )
+      me.getButton( button )
         .classList.remove( color )
       }, 100 )
   }
 
-  window.onkeydown = function( event ) {
-    isNumber( event.key )
-      ? ( fakeClickEffect( event.key, cssClasses.GREY_BUTTON_PRESSED ),
-        enterDigit( event.key ) )
-      : isOperator( event.key )
-      ? ( fakeClickEffect( event.key, cssClasses.ORANGE_BUTTON_PRESSED ),
-        setOperator( event.key ) )
-      : isEquals( event.key )
-      ? ( fakeClickEffect( buttons.EQUALS, cssClasses.ORANGE_BUTTON_PRESSED ),
-        calculate() )
-      : isClear( event.key )
-      ? ( fakeClickEffect( buttons.CLEAR, cssClasses.GREY_BUTTON_PRESSED ),
-        clearCalculator() )
-      : isDot( event.key )
-      ? ( fakeClickEffect( buttons.DOT, cssClasses.GREY_BUTTON_PRESSED ),
-        addDot() )
-      : null
-  }
-
-  function addListeners() {
-    document.querySelector( elementSelectors.BUTTON_SECTION )
+  Calculator.prototype.addListeners = function () {
+    //FIXME: ghetto binding
+    var me = this
+    this.dom.querySelector( Calculator.elementSelectors.BUTTON_SECTION )
```

Comment:
do this within your constructor `this.onButtonClick = this.onButtonClick.bind(this)` and then here just do `.addEventListener( 'click', this.onButtonClick)` and you wont need this `var me = this` hack